examples/benchmarking_comparison_models.py

```diff
@@ -0,0 +1,209 @@
+# object, which gets logged and stored by the system so you can remember how you
+# configured things for a specific model later.
+
+
+# In this case we have a very simple rollout function, but you can see how we're
+# using the `use_thinking` flag to change the prompt slightly.
+async def rollout(model: art.Model, scenario: MyScenarioSpec) -> art.Trajectory:
+    assert isinstance(model.config, MyConfig)
+
+    openai_client = model.openai_client()
+    messages = []
+    if model.config.use_thinking:
+        messages.append(
+            {
+                "role": "system",
+                "content": "You'll be given a math problem by the user. Before answering, first output your thinking inside <thinking> tags. Then, output your answer inside <answer> tags.",
+            }
+        )
+
+    messages.append(
+        {"role": "user", "content": f"What is {scenario.input_1} + {scenario.input_2}?"}
+    )
+
+    response = await openai_client.chat.completions.create(
+        model=model.get_inference_name(),
+        messages=messages,
+    )
+    choice = response.choices[0]
+    response_text = choice.message.content
+    assert response_text is not None
+
+    messages.append(choice)
+
+    # Your rollout function should always return a Trajectory object.
+    return art.Trajectory(
+        messages_and_choices=messages,
+        reward=reward(model.config, scenario, response_text),
+    )
+
+
+async def train_model(model: art.TrainableModel):
+    train_iterator = iterate_dataset(
+        training_scenarios,
+        groups_per_step=4,
+        initial_step=await model.get_step(),
+    )
+
+    for batch, epoch, global_step, epoch_step in train_iterator:
+        groups = await art.gather_trajectory_groups(
+            art.TrajectoryGroup(
+                (rollout(model, scenario) for _ in range(6)),
+            )
+            for scenario in batch
+        )
+        await model.train(groups)
+
+        if global_step % 20 == 0:
+            # Every 20 steps let's benchmark our model under training so we can
+            # see how it's doing.
+            await benchmark_model(model)
+
+    # At the end of training, let's benchmark the model again to see where it
+    # ended up.
+    await benchmark_model(model)
+
+
+async def main():
+    # Here we define a list of prompted comparison models we want to assess
+    # performance for.
+    gpt_4_1 = art.Model(
+        name="gpt-4.1",
+        project=PROJECT_NAME,
+        inference_base_url="https://api.openai.com/v1/",
+        inference_api_key=os.getenv("OPENAI_API_KEY"),
+    )
+
+    # We'll create a second GPT-4.1 variant that uses CoT reasoning
+    gpt_4_1_thinking = gpt_4_1.model_copy(deep=True)
+    gpt_4_1_thinking.name = "gpt-4.1-thinking"
+    assert isinstance(gpt_4_1_thinking.config, MyConfig)
+    gpt_4_1_thinking.config.use_thinking = True
+
+    gemini_flash = gpt_4_1.model_copy(deep=True)
```

Comment:
I realize this is up to the user, but do you see a benefit to copying here over redeclaring a new gemini_flash model from scratch? Seems like the only preserved property is the project name.